File: Second year/Design and Programming Environments/TicketManagerJava/Networking/src/main/java/DTOPackage/DTOUtils.java
package DTOPackage;


import ModelDomain.Event;
import ModelDomain.Ticket;
import ModelDomain.User;

public class DTOUtils {
    public static User getFromDTO(UserDTO userDTO){
        String username=userDTO.getUsername();
        String password=userDTO.getPassword();
        return new User(username,password);
    }

    public static UserDTO getDTO(User user){
        String username=user.getId();
        String password=user.getPassword();
        return new UserDTO(username,password);
    }

    public static Event getFromDTO(EventDTO eventDTO){
        Integer id=eventDTO.getId();
        String gameName=eventDTO.getGameName();
        Double gamePrice=eventDTO.getGamePrice();
        Integer freeSeats=eventDTO.getFreeSeats();
        return new Event(id,gameName,gamePrice,freeSeats);
    }

    public static EventDTO getDTO(Event event){
        Integer id=event.getId();
        String gameName=event.getGameName();
        Double gamePrice=event.getGamePrice();
        Integer freeSeats=event.getFreeSeats();
        return new EventDTO(id,gameName,gamePrice,freeSeats);
    }

    public static TicketDTO getDTO(Ticket ticket){
        Integer id=ticket.getId();
        Integer idGame=ticket.getId_game();
        Integer reservedSeats=ticket.getReservedSeats();
        Double price=ticket.getPrice();
        String clientName=ticket.getClientName();
        return new TicketDTO(id,idGame,reservedSeats,price,clientName);
    }

    public static Ticket getFromDTO(TicketDTO ticketDTO){
        Integer id=ticketDTO.getId();
        Integer idGame=ticketDTO.getIdGame();
        Integer reservedSeats=ticketDTO.getReservedSeats();
        Double price=ticketDTO.getPrice();
        String clientName=ticketDTO.getClientName();
        return new Ticket(id,idGame,reservedSeats,price,clientName);
    }

}

File: Second year/Design and Programming Environments/TicketManagerCSharp/Networking/DTOS/DTOUtils.cs
using TicketManagerCSharp.domain;

namespace Networking.DTOS
{
    public class DTOUtils
    {
        public static User getFromDTO(UserDTO user)
        {
            string username = user.Username;
            string password = user.Password;
            return new User(username, password);
        }

        public static UserDTO getDTO(User user)
        {
            string username = user.Id;
            string password = user.Password;
            return new UserDTO(username, password);
        }

        public static Event getFromDTO(EventDTO ev)
        {
            return new Event(ev.Id, ev.GameName, ev.GamePrice, ev.FreeSeats);
        }

        public static EventDTO getDTO(Event ev)
        {
            return new EventDTO(ev.Id, ev.GameName, ev.GamePrice, ev.FreeSeats);
        }

        public static Ticket getFromDTO(TicketDTO ticket)
        {
            return new Ticket(ticket.Id, ticket.IdGame, ticket.ReservedSeats, ticket.Price, ticket.ClientName);
        }

        public static TicketDTO getDTO(Ticket ticket)
        {
            return new TicketDTO(ticket.Id, ticket.IdGame, ticket.ReservedSeats, ticket.Price, ticket.ClientName);
        }
    }
}